### ISSUE TITLE
Bitrise Steps

### ISSUE THREAD

**Issue by brandtdaniels** [ChargePoint/xcparse#29]
I would love to see some Bitrise steps using StepLib for this

**Comment by abotkin-cpi** [comment]
Regardless of Bitrise's plan for Xcode Test, given #33 it seems there is value in this being a separate Bitrise Step as I can't imagine their Xcode Test step may want to extract everything in the xcresult that some folks would want.  Planning to look at this a bit during the Thanksgiving break & get something up first week of December.

**Comment by abotkin-cpi** [comment]
A working prototype of [xcparse as a Bitrise step](https://github.com/ChargePoint/bitrise-step-xcparse) is available now.  This prototype will extract everything with no options added & put them into ZIP folders whose paths are available as outputs from the step.  If you want to keep the ZIP folders, you'll want to move them to the Deploy folder and make sure you have the Deploy step after.

To add it to your workflow, add the following into your bitrise.yml in the Workflow Editor after the Xcode Test step that generates your xcresult:

```
    - git::https://github.com/ChargePoint/bitrise-step-xcparse.git@master:
        title: xcparse
    - script@1.1.5:
        inputs:
        - content: |-
            #!/usr/bin/env bash
            # fail if any commands fails
            set -e
            # debug log
            set -x

            mv "$XCPARSE_ATTACHMENTS_PATH" "$BITRISE_DEPLOY_DIR"
            mv "$XCPARSE_CODE_COVERAGE_PATH" "$BITRISE_DEPLOY_DIR"
            mv "$XCPARSE_LOGS_PATH" "$BITRISE_DEPLOY_DIR"
            mv "$XCPARSE_SCREENSHOTS_PATH" "$BITRISE_DEPLOY_DIR"
        title: Move xcparse results to Deploy
```

If you don't use the native Xcode Test step, save the following into bitrise.yml & then go back to the GUI workflow editor and change the xcresult path from the environment variable Bitrise has to the path for your xcresult.

After Thanksgiving, we'll expand the step to allow for configuring the various options of xcparse & then submit to Bitrise's step library.  Feature suggestions welcome!

If there are any folks interested in providing feedback or wanting to be beta testers, hit me up by email (check my profile) or by replying here.

**Comment by abotkin-cpi** [comment]
The Bitrise step has been updated to allow for much more flexibility from the proof of concept & we've submitted a request to get it merged into the Bitrise Steplib so that it can be easily added from the Bitrise Workflow UI (bitrise-io/bitrise-steplib#2306).

**Comment by abotkin-cpi** [comment]
Just providing an update - we're still waiting for feedback/review from Bitrise maintainers & this is blocked on the earlier linked PR.

**Comment by abotkin-cpi** [comment]
As of today, xcparse is now an available step in Bitrise's StepLib & can be added directly in Bitrise.io's Workflow Editor!  See the [README of bitrise-step-xcparse](https://github.com/ChargePoint/bitrise-step-xcparse) for more information.

Let us know in the Issues over in that repo if there's any other features or enhancements you'd like to see & thanks for the feature suggestion @brandtdaniels!
